src/transactions/SetTrustLineFlagsOpFrame.h

```diff
@@ -4,34 +4,54 @@
 // under the Apache License, Version 2.0. See the COPYING file at the root
 // of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
 
-#include "transactions/OperationFrame.h"
+#include "transactions/TrustlineOpFrameBase.h"
 
 namespace stellar
 {
 class AbstractLedgerTxn;
 
-class SetTrustLineFlagsOpFrame : public OperationFrame
+class SetTrustLineFlagsOpFrame : public TrustlineOpFrameBase
 {
-    ThresholdLevel getThresholdLevel() const override;
     SetTrustLineFlagsResult&
     innerResult()
     {
         return mResult.tr().setTrustLineFlagsResult();
     }
 
-    bool isAuthRevocationValid(AbstractLedgerTxn& ltx);
-
     SetTrustLineFlagsOp const& mSetTrustLineFlags;
 
     uint32_t mOpIndex;
 
+    inline void setResultSelfNotAllowed() override{};
```

Comment:
Thanks for the inline suggestion. In that case is it ok to have the one-liner implementation (the `throw` statement) in the header file (and will be implicitly treated as inline), or always move it to the cpp?

Reply:
Either is fine, we have a mixture across the code base. But I personally have a preference for putting all implementations in the cpp file, because it (a) means I always know where to look for the implementation and (b) helps keep compile times down if I have to modify the implementation--for example, if we wanted to change the error message.